src/main/java/io/stargate/sgv2/jsonapi/service/schema/tables/ApiIndexType.java

```diff
@@ -53,31 +53,29 @@ public static ApiIndexType fromTypeName(String typeName) {
   static ApiIndexType fromCql(
       ApiColumnDef apiColumnDef, CQLSAIIndex.IndexTarget indexTarget, IndexMetadata indexMetadata)
       throws UnsupportedCqlIndexException {
-
-    // If there is no function on the indexTarget, and the column is a scalar, then it is a regular
-    // index on
-    // an int, text, etc
-    if (indexTarget.indexFunction() == null && apiColumnDef.type().isPrimitive()) {
+    // If cqlIndexType is values, and the column is a scalar
+    // then it is a regular index on primitive types
+    if (indexTarget.cqlIndexType() == CQLSAIIndex.CqlIndexType.VALUES
```

Comment:
this feels risky - we are inferring the VALUES function when it is not defined in the CQL for the index and then using that inferred value. The previous code did not infer a function when there was none defined.